### DISCUSSION TITLE
How to model the link between the VM guest interfaces and their host counterpart ?

### DISCUSSION THREAD

**Discussion by rgrunbla** [netbox-community/netbox#11981] [Q&A]
Hello,

I'm using netbox to model my infrastructure which is using proxmox (and other things, of course). VMs on proxmox have interfaces which correspond to tap interfaces on the host, e.g. eth0 (in the VM) and tap100i0 (on the host). I'd like to model this in Netbox, for example with a virtual cable, but it seems I cannot create connections with virtual interfaces, e.g. the documentation states that « only physical interfaces may be connected via cables ». As I do not have the option (it seems) to mark the VM interfaces as physical, what are my option to say that interface X on the host corresponds to interface Y in the VM ?

Thanks !

**Comment by a084ed22** [comment]
Is there a compelling reason to match the interfaces of the VMs with their counterpart on the physical host? I don't think it is stable iow it will map to different names over reboots.

**Reply by rgrunbla** [reply]
For proxmox, it actually is, because the tap interfaces follow the `tap<vmid>i<netnum>` convention and the VM ID or Net Number do not change.

**Comment by mtinberg** [comment]
Can you just create a single object custom field on /virtualization/vm_interface pointing to /dcim/interface to document this relationship?


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Rémy Grünblatt ***@***.***>
Sent: Wednesday, March 15, 2023 7:15 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] How to model the link between the VM guest interfaces and their host counterpart ? (Discussion #11981)


Hello,

I'm using netbox to model my infrastructure which is using proxmox. VMs on proxmox have interfaces which correspond to tap interfaces on the host, e.g. eth0 (in the VM) and tap100i0 (on the host). I'd like to model this in Netbox, for example with a virtual cable, but it seems I cannot create connections with virtual interfaces, e.g. the documentation states that « only physical interfaces may be connected via cables ». As I do not have the option (it seems) to mark the VM interfaces as physical, what are my option to say that interface X on the host corresponds to interface Y in the VM ?

Thanks !
